server/parameters/CoreSubcommandParser.java

```diff
@@ -34,34 +34,35 @@
 import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_UNRECOGNISED;
 import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
 
-public class ServerSubcommandParser {
+public class CoreSubcommandParser {
 
-    private static void validateRequiredOptions(Set<OptionParser> requiredParsers, Set<Option> options) {
+    protected static void validateRequiredOptions(Set<OptionParser> requiredParsers, Set<Option> options) {
         requiredParsers.forEach(parser -> {
             if (iterate(options).noneMatch(option -> option.name().equals(parser.name()))) {
                 throw TypeDBException.of(CLI_OPTION_REQUIRED, parser.name());
             }
         });
     }
 
-    private static void validateUnrecognisedOptions(Set<OptionParser> recognisedParsers, Set<Option> options) {
-        recognisedParsers.forEach(parser -> {
-            if (iterate(options).noneMatch(option -> option.name().equals(parser.name()))) {
-                throw TypeDBException.of(CLI_OPTION_UNRECOGNISED, parser);
+    protected static void validateUnrecognisedOptions(Set<OptionParser> recognisedParsers, Set<Option> options) {
+        options.forEach(option -> {
+            if (iterate(recognisedParsers).noneMatch(parser -> option.name().equals(parser.name()))) {
+                throw TypeDBException.of(CLI_OPTION_UNRECOGNISED, option.name());
             }
         });
     }
 
-    public static class Server extends SubcommandParser<ServerSubcommand.Server> {
+    public static class Server extends SubcommandParser<CoreSubcommand.Server> {
+
+        public static final String[] tokens = new String[]{};
+        public static final String description = "Run TypeDB server";
 
-        private static final OptionParser.Flag debug = new OptionParser.Flag("debug", "Run server in debug mode.");
-        private static final OptionParser.Flag help = new OptionParser.Flag("help", "Print help menu.");
-        private static final OptionParser.Flag version = new OptionParser.Flag("version", "Print version number.");
-        private static final OptionParser.Path configPath =
+        public static final OptionParser.Flag debug = new OptionParser.Flag("debug", "Run server in debug mode.");
```

Comment:
we make these `public` so that Cluster can re-use them without extending the class directly.